src/util/test/date-and-time-test.js

```diff
@@ -0,0 +1,279 @@
+        options: { day: 'numeric', month: 'short' },
+      },
+      {
+        now: '1970-07-03T00:00:00.000Z',
+        text: '3 July',
+        options: { day: 'numeric', month: 'short' },
+      },
+      {
+        now: '1971-01-01T00:00:00.000Z',
+        text: '1 Jan 1970',
+        options: { day: 'numeric', month: 'short', year: 'numeric' },
+      },
+      {
+        now: '1971-03-01T00:00:00.000Z',
+        text: '1 Jan 1970',
+        options: { day: 'numeric', month: 'short', year: 'numeric' },
+      },
+      {
+        now: '1972-01-01T00:00:00.000Z',
+        text: '1 Jan 1970',
+        options: { day: 'numeric', month: 'short', year: 'numeric' },
+      },
+      {
+        now: '1978-01-01T00:00:00.000Z',
+        text: '1 Jan 1970',
+        options: { day: 'numeric', month: 'short', year: 'numeric' },
+      },
+    ].forEach(test => {
+      it(
+        'passes correct arguments to `Intl.DateTimeFormat.format` for fixture ' +
+          test.now,
+        () => {
+          const timeStamp = fakeDate('1970-01-01T00:00:00.000Z');
+          const now = fakeDate(test.now);
+
+          fakeIntl.DateTimeFormat().format.returns(test.text); // eslint-disable-line new-cap
+          assert.equal(formatRelativeDate(timeStamp, now, fakeIntl), test.text);
+          assert.calledWith(fakeIntl.DateTimeFormat, undefined, test.options);
+          assert.calledWith(fakeIntl.DateTimeFormat().format, timeStamp); // eslint-disable-line new-cap
+        },
+      );
+    });
+  });
+
+  describe('decayingInterval', () => {
+    it('handles empty dates', () => {
+      const date = null;
+      decayingInterval(date, undefined);
+    });
+
+    it('never invokes callback if date is invalid', () => {
+      const date = new Date('foo bar');
+      const callback = sinon.stub();
+
+      decayingInterval(date, callback);
+      sandbox.clock.tick(600 * day);
+
+      assert.notCalled(callback);
+    });
+
+    it('uses a short delay for recent timestamps', () => {
+      const date = new Date().toISOString();
+      const callback = sandbox.stub();
+      decayingInterval(date, callback);
+      sandbox.clock.tick(6 * second);
+      assert.called(callback);
+      sandbox.clock.tick(6 * second);
+      assert.calledTwice(callback);
+    });
+
+    it('uses a longer delay for older timestamps', () => {
+      const date = new Date().toISOString();
+      const ONE_MINUTE = minute;
+      sandbox.clock.tick(10 * ONE_MINUTE);
+      const callback = sandbox.stub();
+      decayingInterval(date, callback);
+      sandbox.clock.tick(ONE_MINUTE / 2);
+      assert.notCalled(callback);
+      sandbox.clock.tick(ONE_MINUTE);
+      assert.called(callback);
+      sandbox.clock.tick(ONE_MINUTE);
+      assert.calledTwice(callback);
+    });
+
+    it('returned function cancels the timer', () => {
+      const date = new Date().toISOString();
+      const callback = sandbox.stub();
+      const cancel = decayingInterval(date, callback);
```

Comment:
We could modernize this API in future by returning an `AbortSignal`. This would make the meaning of the return value more obvious and potentially simplify use in other browser APIs.